src/main/java/com/atlassian/bitbucket/jenkins/internal/client/BitbucketRequestExecutor.java

```diff
@@ -108,27 +109,28 @@ public void makeDeleteRequest(HttpUrl url) {
      * Makes a POST request to the given URL with given request payload.
      *
      * @param url             the URL to make the request to
-     * @param requestPayload, JSON payload which will be marshalled to send it with POST.
+     * @param requestPayload, JSON payload which will be marshalled to send it with POST
      * @param returnType,     class of expected return type
-     * @param <T>             type of Request payload.
+     * @param <T>             type of Request payload
      * @param <R>             return type
      * @return the result
      */
-    public <T, R> BitbucketResponse<R> makePostRequest(HttpUrl url, T requestPayload, Class<R> returnType) {
+    public <T, R> BitbucketResponse<R> makePostRequest(HttpUrl url, T requestPayload, Map<String, String> headers,
+                                                       Class<R> returnType) {
         ObjectReader<R> reader = in -> objectMapper.readValue(in, returnType);
         return httpRequestExecutor.executePost(url, credentials, marshall(requestPayload), response ->
-                new BitbucketResponse<>(response.headers().toMultimap(), unmarshall(reader, response.body())));
+                new BitbucketResponse<>(response.headers().toMultimap(), unmarshall(reader, response.body())), headers);
     }
 
     /**
      * Makes a POST request to the given URL with given request payload.
      *
      * @param url            the URL to make the request to
-     * @param requestPayload JSON payload which will be marshalled to send it with POST.
-     * @param <T>            Type of Request payload.
+     * @param requestPayload JSON payload which will be marshalled to send it with POST
+     * @param <T>            Type of Request payload
      */
-    public <T> void makePostRequest(HttpUrl url, T requestPayload) {
-        httpRequestExecutor.executePost(url, credentials, marshall(requestPayload), EMPTY_RESPONSE);
+    public <T> void makePostRequest(HttpUrl url, T requestPayload, Map<String, String> headers) {
```

Comment:
Minor: given the couple of cases where we're passing in empty maps to this method, would it be worth adding an overloaded method for that?
```
public <T> void makePostRequest(HttpUrl url, T requestPayload) {
    makePostRequest(HttpUrl url, T requestPayload, Collections.emptyMap());
}
```

Reply:
That's what I had initially, there's another review comment from Gaurav suggesting I should remove the method. I don't really mind either way.

Reply:
Ah, fair. No worries then :+1: